File: app/src/main/java/com/mhammad/photoapp/api/Hit.java
package com.mhammad.photoapp.api;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "posts_table")
public class Hit {
    public Hit(String largeImageURL, int views, int likes, int comments, String user, String userImageURL) {
        this.largeImageURL = largeImageURL;
        this.views = views;
        this.likes = likes;
        this.comments = comments;
        this.user = user;
        this.userImageURL = userImageURL;
    }
@PrimaryKey(autoGenerate = true)
    private  int _ID;
    public int id;
    public String pageURL;
    public String type;
    public String tags;
    public String previewURL;
    public int previewWidth;
    public int previewHeight;
    public String webformatURL;
    public int webformatWidth;
    public int webformatHeight;
    public String largeImageURL;
    public int imageWidth;
    public int imageHeight;
    public int imageSize;
    public int views;
    public int downloads;
    public int collections;
    public int likes;
    public int comments;
    public int user_id;
    public String user;
    public String userImageURL;

    public int get_ID() {
        return _ID;
    }

    public void set_ID(int _ID) {
        this._ID = _ID;
    }
}

File: app/src/main/java/com/example/pixabay/model/Hit.kt
package com.example.pixabay.model

import androidx.databinding.Bindable
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "posts")
data class Hit(
    val collections: Int,
    val comments: Int, //# of comments of the photo
    val downloads: Int, //# of downloads of the photo
    @PrimaryKey(autoGenerate = false)
    @ColumnInfo(name = "post_id")
    val id: Int,
    val imageHeight: Int,
    val imageSize: Int,
    val imageWidth: Int,
    val largeImageURL: String,
    val likes: Int, //# of likes of photoes that received from the API
    val pageURL: String,
    val previewHeight: Int,
    val previewURL: String,
    val previewWidth: Int,
    val tags: String,
    val type: String,
    val user: String, //user_name
    val userImageURL: String,
    val user_id: Int,
    val views: Int,// # of views
    val webformatHeight: Int,
    val webformatURL: String,
    val webformatWidth: Int,
    var isFavourite: Boolean
)